Make possible use docker compose and make possible change the source repository

vitormattos opened the pull request.

vitormattos committed:
Make possible use custom MTProxy project

Signed-off-by: Vitor Mattos <vitor@php.rio>

vitormattos committed:
Add more instructions about this project

Signed-off-by: Vitor Mattos <vitor@php.rio>

vitormattos committed:
Add information about the fork from MTProto project

Signed-off-by: Vitor Mattos <vitor@php.rio>

Dofamin commented on Dockerfile:
```diff
@@ -6,10 +6,12 @@ LABEL image="MTProxy"
+ARG WORKERS
+ENV WORKERS=${WORKERS:-1}
+ARG MTPROTO_REPO_URL
+ENV MTPROTO_REPO_URL=${MTPROTO_REPO_URL:-https://github.com/GetPageSpeed/MTProxy}
```
by default this project use official repository, leave him as default
https://github.com/TelegramMessenger/MTProxy

Dofamin commented on Dockerfile:
```diff
@@ -6,10 +6,12 @@ LABEL image="MTProxy"
+ARG MTPROTO_REPO_URL
+ENV MTPROTO_REPO_URL=${MTPROTO_REPO_URL:-https://github.com/GetPageSpeed/MTProxy}
 WORKDIR /srv/
 ENV TZ=Europe/Moscow
```
You declare Environment TZ, but it is static at the moment

Dofamin commented on docker-compose.yml:
```diff
@@ -0,0 +1,20 @@
+      - 8443:8889/udp
+      - 8443:8889/tcp
+    environment:
+      - TAG
```
What is this ENV?

vitormattos replied:
Removed.

Dofamin commented on docker-compose.yml:
```diff
@@ -0,0 +1,20 @@
+    environment:
+      - TAG
+      - SECRET
+      - SECRET_COUNT
```
What is this ENV?

vitormattos replied:
Sorry, I think that is an old copy past. I implemented this have a lot of time but never did the PR. Looking the code sounds that is unused environments, except the `SECRET`. I  removed all unused.

Dofamin commented on README.md:
```diff
@@ -111,4 +117,4 @@ If clients hang in an "updating" state, be sure to check the following:
 
 ##### [Official MTProxy GitHub Repository](https://github.com/TelegramMessenger/MTProxy)
-
+##### [Forked project from MTProxy with patches](https://github.com/GetPageSpeed/MTProxy)
```
```suggestion
add as alternatives
```

vitormattos replied:
Done, I documented a bit more the environments with this second repository as an usage example.

vitormattos committed:
Define the original project as default repo url

Defined as default the original project and added a mention about the
fork that have patches applied because the original project is a bit
outdated.

Signed-off-by: Vitor Mattos <vitor@php.rio>

vitormattos committed:
Moved environments to uppercase

Uppercase is a pattern used by a lot of languages and systems to define
an environment variable. Replaced the patter with upper case first to
all in uppercase.

Signed-off-by: Vitor Mattos <vitor@php.rio>

vitormattos committed:
Removed unused environments

Signed-off-by: Vitor Mattos <vitor@php.rio>
